cellprofiler/modules/watershed.py

```diff
@@ -407,179 +411,59 @@ def run(self, workspace):
-            strel_dim = self.structuring_element.value.ndim
-
-            # test if the structuring element dimensions match the image dimensions
-            if strel_dim != dimensions:
-                raise ValueError("Structuring element does not match object dimensions: "
-                                 "{} != {}".format(strel_dim, dimensions))
-
-            # Get the segmentation distance transform for the watershed segmentation
-            peak_image = scipy.ndimage.distance_transform_edt(y_data > 0)
-
-            # shape-based method; generate a watershed ready image
-            if self.declump_method.value == O_SHAPE:
-                # Use the reverse of the image to get basins at peaks
-                watershed_image = -peak_image
-                watershed_image -= watershed_image.min()
-
-            # intensity-based method
-            else:
-                # get the intensity image data
-                reference_name = self.reference_name.value
-                reference = images.get_image(reference_name)
-                reference_data = reference.pixel_data
-
-                # Set the image as a float and rescale to full bit depth
-                watershed_image = skimage.img_as_float(reference_data, force_copy=True)
-                watershed_image -= watershed_image.min()
-                watershed_image = 1 - watershed_image
-
-                if reference.multichannel:
-                    watershed_image = skimage.color.rgb2gray(watershed_image)
-
-            # Smooth the image
-            watershed_image = skimage.filters.gaussian(watershed_image, sigma=self.gaussian_sigma.value)
-
-            # Generate local peaks; returns a list of coords for each peak
-            seed_coords = skimage.feature.peak_local_max(peak_image,
-                                                   min_distance=self.min_dist.value,
-                                                   threshold_rel=self.min_intensity.value,
-                                                   exclude_border=self.exclude_border.value,
-                                                   num_peaks=self.max_seeds.value if self.max_seeds.value != -1 else numpy.inf)
-
-            # generate an array w/ same dimensions as the original image with all elements having value False
-            seeds = numpy.zeros_like(peak_image, dtype=bool)
-
-            # set value to True at every local peak
-            seeds[tuple(seed_coords.T)] = True
-
-            # Dilate seeds based on settings
-            seeds = skimage.morphology.binary_dilation(seeds, self.structuring_element.value)
-
-            # get the number of objects from the distance-based or marker-based watershed run above
-            number_objects = skimage.measure.label(y_data, return_num=True)[1]
-
-            seeds_dtype = (numpy.uint16 if number_objects < numpy.iinfo(numpy.uint16).max else numpy.uint32)
-
-            # NOTE: Not my work, the comments below are courtesy of Ray
```

Comment:
I haven't tested it, but it's probably worth doing so to see if this "make things negative, then back to positive again" trick is still useful/helpful with the 2023 versions of these algorithms, and if so, taking what's in ~rows 553-579 and moving it into the new function.